dbms/src/Interpreters/SharedContexts/Disagg.cpp

```diff
@@ -105,4 +105,14 @@ void SharedContextDisagg::initFastAddPeerContext(UInt64 fap_concur)
     fap_context = std::make_shared<FastAddPeerContext>(fap_concur);
 }
 
+
+SharedContextDisagg::~SharedContextDisagg()
+{
+    LOG_INFO(Logger::get(), "Shutdown SharedContextDisagg");
+    if (fap_context)
+    {
+        fap_context->shutdown();
```

Comment:
should we call `fap_context.reset()` after shutdown?

Reply:
I think it's necessary. IMO, We don't need to eagerly reclaim the `fap_context`, we just need to eagerly join all workers in AsyncTasks.